Tests/Realm.Tests/Database/NotificationTests.cs

```diff
@@ -195,6 +195,179 @@ public void DictionaryNotificationToken_KeepsCollectionAlive()
+            _realm.Write(() =>
+            {
+                list.First().Nickname = "foo1";
+                set.First().Nickname = "foo2";
+                dictionary.First().Value.Nickname = "foo3";
+            });
+            _realm.Refresh();
+
+            // Moving
+            _realm.Write(() =>
+            {
+                list.Move(0, 1);
+            });
+            _realm.Refresh();
+
+            // Removing
+            _realm.Write(() =>
+            {
+                _realm.Remove(list.First());
+                _realm.Remove(set.First());
+                dictionary.Remove("foo");
+            });
+            _realm.Refresh();
+        }
+
+        [Test]
+        public void CollectionPropertiesOfSameTypeShouldNotFireNotificationsOnChange()
+        {
+            var testObject = _realm.Write(() => _realm.Add(new TestNotificationObject()));
+            testObject.PropertyChanged += (sender, e) =>
+            {
+                Assert.That(e.PropertyName, Is.EqualTo("StringProperty"));
+            };
+
+            var testObject1 = new TestNotificationObject();
+            var testObject2 = new TestNotificationObject();
+            var testObject3 = new TestNotificationObject();
+            var testObject4 = new TestNotificationObject();
+            var testObject5 = new TestNotificationObject();
+            var list = testObject.ListSameType;
+            var set = testObject.SetSameType;
+            var dictionary = testObject.DictionarySameType;
+
+            // Insertions
+            _realm.Write(() =>
+            {
+                list.Add(testObject1);
+                set.Add(testObject2);
+                dictionary.Add("foo", testObject3);
+            });
+
+            // Update primitive property while inserting
+            _realm.Write(() =>
+            {
+                // Only this assignment should fire a notification
+                testObject.StringProperty = "foo";
+                list.Add(testObject4);
+                list.Add(testObject5);
+            });
+            _realm.Refresh();
+
+            // Modifications
+            _realm.Write(() =>
+            {
+                list.First().StringProperty = "foo1";
+                set.First().StringProperty = "foo2";
+                dictionary.First().Value.StringProperty = "foo3";
+            });
+            _realm.Refresh();
+
+            // Moving
+            _realm.Write(() =>
+            {
+                list.Move(0, 1);
+            });
+            _realm.Refresh();
+
+            // Removing
+            _realm.Write(() =>
+            {
+                _realm.Remove(list.First());
+                _realm.Remove(set.First());
+                dictionary.Remove("foo");
+            });
+            _realm.Refresh();
+        }
+
+        [Test]
+        public void Link_ShouldOnlyFireNotificationForReassignment()
+        {
+            var testObject = new TestNotificationObject();
```

Comment:
I think you should test if `StringProperty`'s notifications fire properly also here.